### PR TITLE
Xli/cognito

### PR DESCRIPTION
**Pull request by sinnyn**
Cognito integration

### CODE REVIEW

**Comment by sinnyn** [comment]
1. Integrates AWS cognito
2. Redirect to login page if validation failed
3. Redirect to login page if token is expired
4. Handle tokens in oauth way

File: package.json [hunk 1]
```diff
@@ -1,161 +1,177 @@
 {
-    "name": "scalabel",
-    "repository": {
-        "type": "git",
-        "url": "https://github.com/ucbdrive/scalabel"
+  "name": "scalabel",
```
Comment (fyu) [inline package.json:2]: please keep 4 space indent for json

**Comment by fyu** [comment]
Please add screenshots of the process to illustrate the steps.

**Comment by sinnyn** [comment]
Entry page
WX20200420-133513@2x
Click open/create -> If session not exists, page jump to auth page.
WX20200420-133539@2x
If auth passes it will jump to callback page, after token set in the local storage, it will redirect to the create page, or in last step session exists, go straight to the create page.
WX20200420-133604@2x

File: app/src/js/server/types.ts [hunk 2]
```diff
@@ -47,6 +47,16 @@ export interface ServerConfig {
   botHost: string
   /** port of python model server */
   botPort: number
+  /** region of cognito service */
```
[the remark this answers is not in the document]
Comment (sinnyn) [reply app/src/js/server/types.ts]: See updated commit

File: app/src/js/common/service.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import { CognitoAccessToken, CognitoAuthSession, CognitoIdToken, CognitoRefreshToken } from 'amazon-cognito-auth-js'
```
[the remark this answers is not in the document]
Comment (sinnyn) [reply app/src/js/common/service.ts]: See updated commit

File: app/src/js/server/util.ts [hunk 4]
```diff
@@ -56,20 +56,26 @@ export function readConfig (): ServerConfig {
...
-    if (!Object.prototype.hasOwnProperty.call(config, 'clientId')) {
-      throw new Error('Client id missed in config')
-    }
-    if (!Object.prototype.hasOwnProperty.call(config, 'userPoolBaseUri')) {
-      throw new Error('User pool base uri missed in config')
-    }
-    if (!Object.prototype.hasOwnProperty.call(config, 'callbackUri')) {
-      throw new Error('Call back uri missed in config')
+    if (config.cognito) {
+      if (!Object.prototype.hasOwnProperty.call(config.cognito, 'region')) {
```
[the remark this answers is not in the document]
Comment (sinnyn) [reply app/src/js/server/util.ts]: Revised

**Pull request merged**